### PR TITLE
[cling] Set up TargetMachine for JITLink and simplify its creation

### PR DESCRIPTION
**Pull request by hahnjo**

### CODE REVIEW

**Comment by hahnjo** [comment]
FWIW, it's possible to run the full tests with `CLING_JITLINK=1` on Linux x86_64 with this, and get "only" 48 failures :tada:

File: interpreter/cling/lib/Interpreter/IncrementalJIT.h [hunk 1]
```diff
@@ -109,6 +109,7 @@ class IncrementalJIT {
   std::map<const Transaction*, llvm::orc::ResourceTrackerSP> m_ResourceTrackers;
   std::map<const llvm::Module *, llvm::orc::ThreadSafeModule> m_CompiledModules;
 
+  bool JITLink;
```
Comment (vgvassilev) [inline interpreter/cling/lib/Interpreter/IncrementalJIT.h:112]: Can you prefix this with m_

**Review by vgvassilev — COMMENTED** [review]
Looks good to me. What’s the state of cling’s test suite?

**Comment by Axel-Naumann** [comment]
> What’s the state of cling’s test suite?

I can answer: failing since [Thu Feb 25 2022](https://lcgapp-services.cern.ch/root-jenkins/view/cling/job/cling-generic-build/3113/), i.e. *almost* a year.

**Comment by hahnjo** [comment]
Yes, and just to confirm, it's not getting worse with this PR.

File: interpreter/cling/lib/Interpreter/IncrementalJIT.h (hunk 1 above)
Comment (hahnjo) [reply interpreter/cling/lib/Interpreter/IncrementalJIT.h]: Done, also renamed `m_TM`.

**Review by vgvassilev — APPROVED** [review]
LGTM!

**Pull request merged**